### PR TITLE
Major and minor memory leak fixes + server crash exploit

### PR DESCRIPTION
**Pull request by krisdgreat**
* Re-added sanity checks on mcommandparameterstring/blob
* Quick fix for memory leak - replace MakeParam with new()

Memory leak caused by MakeParam template skipping CMemPool::operator new, combined with the destructor inheritance on MCommandParameters - CMemPool::operator delete is called instead of deallocate or ::delete.
Therefore attempting to un-link memory that was never linked to the mempool in the first place.

Bottom line, MakeParam allocation and construction + the way MCommandParameters are destroyed = not re-using the memory, and not freeing it either

### CODE REVIEW

**Review by Michael-Steshenko — CHANGES_REQUESTED** [review]
make sure SetData not allocating pParam when SetData return false. If null gets deleted no harm done, but if we allocate memory for pParam and we return false, it might not get deleted in the calling function.

remove the break after return call in line 238
